lib/kubeclient/common.rb

```diff
@@ -205,12 +205,13 @@ def retrieve_all_entities(entity_types)
       end
 
       def proxy_url(kind, name, port, namespace = '')
+        entity_name_plural = self.class.pluralize_entity(kind.to_s)
```

Comment:
what if the user will pass  a capitalized version of kind? Pod/Service instead of pod/service?